bitbots_vision/bitbots_vision/vision_modules/yoeo_handler.py

```diff
@@ -0,0 +1,207 @@
+from candidate import CandidateFinder, Candidate
+
+from abc import ABC, abstractmethod
+from collections import defaultdict
+import cv2
+from numpy import where, moveaxis
+from os.path import join
+from rclpy import logging
+from typing import List
+
+logger = logging.get_logger('vision_yoeo')
+
+try:
+    from yoeo import models as torch_models, detect as torch_detect
+    from yoeo.utils.utils import load_classes
+except ImportError:
+    logger.error("Not able to import pytorchyolo. This might be fine if you use another method.")
+
+
+class YOEOHandlerTemplate(ABC):
```

Comment:
I'm not a great fan of having a long inheritance pyramid tbh, especially not with abstract classes. Also, I couldn't even call `super().__init__(...)` for the `YoloHandler` as the it loads the class names "the yolo-way". Maybe it would be a better solution to define a common interface, from which both handlers inherit and which they implement. However, it's also a "not so nice" solution since YOEO would need a more advanced interface than YOLO (or two)...

Reply:
Yea I agree with you in that point.